js-jasmine/src/shop.js

```diff
@@ -1,69 +1,86 @@
 class Shop {
   constructor(items=[]){
     this.items = items;
+    this.count = 0;
   }
-  updateQuality() {
-    for (var i = 0; i < this.items.length; i++) {
-      if (this.checkAgedBrie(i) && this.checkConcert(i)) {
-        if (this.checkqualitygreaterthan0(i) && this.checkHand(i)) {
-            this.changequality(i, -1);
-        }
-      } else {
-        if (this.checkqualitylessthan50(i)) {
-          this.changequality(i, 1);
-          if (!(this.checkConcert(i))&& this.items[i].sellIn < 11) {
-            if (this.items[i].sellIn < 6) {
-              this.changequality(i, -1);
-            }else{
-              this.changequality(i, 1);
-            }
-          }
-        }
-      }
-      if (this.checkHand(i)) {
-        this.items[i].sellIn -= 1;
-      }
-      if (this.items[i].sellIn < 0) {
-        if (this.checkAgedBrie(i)) {
-          if (this.checkConcert(i)) {
-            if (this.checkqualitygreaterthan0(i)) {
-              if (this.checkHand(i)) {
-                this.changequality(i, -1);
-              }
-            }
-          } else {
-            this.items[i].quality = this.items[i].quality - this.items[i].quality;
-          }
-        } else if(this.checkqualitylessthan50(i)) {
-          this.changequality(i, 1);
-        }
-      }
-    }
+  quality(){
+    let templistSlice = this.items;
+    this.updateQuality(templistSlice);
     return this.items;
   }
 
-  checkAgedBrie(i){
-    return this.items[i].name != 'Aged Brie'
+  updateQuality(templist) {
+    if(templist.length === 0){
```

Comment:
I like that you refactored the code into methods, and it is pretty clear what each of them are doing. To take it even further you could refactor it into separate classes as at the moment the code looks quite long and there's a lot going on.

Reply:
i agree with separating into classes to make it even clearer